### PR TITLE
Correctly compute ultra-zoomed line paths [fixes #217]

### LINKED ISSUE #217: Line paths are not correctly computed when their points are well-beyond the axis ranges
When building a scatter plot with both x and y logarithmic axis there is an error when placing a line shape on the plot.

The line is sometimes well displayed but when trying to zoom it disappears or appears not well placed.

The code used for displaying the line is the following:

- inside the layout variable for the plot making place the following:

var layout2 = {
			  .
                          .
                          .
			  shapes: [
			           
					//line 45º
					
					{
					  type: 'line',
					  xref: 'x',
					  yref: 'y',
					  x0: 0.0001,
					  y0: 0.0001,
					  x1: 10000,
					  y1: 10000,
					  line: {
					    color: 'rgb(0, 0, 0)',
					    width: 1
					  }
					},
			           
			           
			           
			           ],
     
				  .
				  .
				  .
			};

there are around 1000 points to be displayed on the scatter plot

Do any of you know if this is a bug or if is is not the proper way to place a line on the plot and there is another method which could work?

Thank you.

### PR DESCRIPTION
**Pull request by etpinard**
@alexcjohnson @mdtusz 

Issue https://github.com/plotly/plotly.js/issues/217 is two months old now and after numerous attempts at fixing it the *right way* by adding more complexity to the already-complex [line decimation](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/line_points.js) algorithm, I gave up.

This PR removes the clip-multiple constrain in the linear to pixel convert routine. This has three effects:

1. Ultra-zoomed line paths (where 4 segment pts are beyond the clip-multiple constrain) are now computed correctly, fixing #217.
1. Far-off-edge points now have coordinates of very large magnitude that some browser's svg renderers **may not** understand. Note: I couldn't find any documentation on the svg limits.
1. The line decimation algorithm is slowed down as now far-off-edge points are not de facto clusters anymore.

I decided to make this PR because I thought that fixing #217 was more important than side-effects 2. and 3. above. I hope you agree.

### CODE REVIEW

File: test/image/mocks/ultra_zoom.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+    "data": [
```
Comment (etpinard) [inline test/image/mocks/ultra_zoom.json]: this mock looks like: 

image

on `master`.

**Comment by mdtusz** [comment]
I did some researching on this before as well and came to the same conclusions (unless we were to implement a much more complex truncate-at-the-plot-borders line decimation method). I couldn't find anything on maximal svg values either - I would think that svg rendering can accomodate arbitrarily large numbers and interpolate _what_ to render if a coordinate lies outside those bounds. 

In any case, if we _do_ encounter issues with this, it will be easy enough to revert to the old solution, or create a more robust one instead. 

:dancer:

**Comment by alexcjohnson** [comment]
Sure. Actually it seems like the way [`getTolerance`](https://github.com/plotly/plotly.js/blob/master/src/traces/scatter/line_points.js#L60-L64) is implemented, the points will continue to get sparser and sparser as they get farther away, so it's likely this won't have *too* much effect on performance anyway. :dancer:

**Comment by alexcjohnson** [comment]
At some point it may be worthwhile to look into making `getTolerance` more aggressive (superlinear in distance from the viewport, perhaps, at least to a point) but this will need some heavy testing to make sure it doesn't introduce new failure modes.

**Pull request merged**